BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/single_file/single_file.rb

```diff
@@ -80,4 +80,25 @@ def gzip_content(operation = :compress) #:nodoc:
     self.sync_to_provider
   end
 
+  # compute file md5 hash sum (useful to avoid tampering, e.g. with task archive)
+  def cache_compute_md5sum
+    md5command =
+        case CBRAIN::System_Uname
+        when /Linux/i
+          "md5sum"
+        when /Darwin/i
+          "md5"
+        else
+          "md5sum"
+        end
+    self.sync_to_cache
+
+    content_path = self&.cache_full_path
+    cb_error("Can't find cache path for file #{self.name}") if content_path.blank?
+    md5 = IO.popen("#{md5command} < #{content_path.to_s.bash_escape}","r") { |fh| fh.read }
+    md5 = Regexp.last_match[1] if md5.present? && md5.match(/\b([0-9a-fA-F]{32})\b/)
+    cb_error("Can't compute MD5 for file #{self.name}", Errno::EIO) if md5.blank? ||
```

Comment:
Also I'm not sure why you wrote that, but cb_error does NOT take a second argument like that. I tried it in my console:

```
CBRAIN> cb_error("Oh no", Errno::EIO)
NoMethodError: undefined method `[]' for Errno::EIO:Class
```

So this means you never exercised that code.

Reply:
done